sdk/identity/azure-identity/src/private/token_credential_impl.hpp

```diff
@@ -59,6 +59,31 @@ namespace Azure { namespace Identity { namespace _detail {
         bool asResource,
         bool urlEncode = true);
 
+    /**
+     * @brief Parses JSON that contains access token and its expiration.
+     *
+     * @param jsonString String with a JSON object to parse.
+     * @param accessTokenPropertyName Name of a property in the JSON object that represents access
+     * token.
+     * @param expiresInPropertyName Name of a property in the JSON object that represents token
+     * expiration in number of seconds from now.
+     * @param expiresOnPropertyNames Names of properties in the JSON object that represent token
+     * expiration as absolute date-time stamp. Can be empty, in which case no attempt to parse the
+     * corresponding property will be made. Empty string elements will be ignored.
```

Comment:
> Can be empty, in which case no attempt to parse the corresponding property will be made.

Is that comment true? I thought if the `expiresOnPropertyNames` vector was empty, this `ParseToken` function would throw.

Reply:
The comment is true. The function is internal. Your implementation does not expect any property to be there - you pass an empty value. It's not really a user input or server response, nor is it a user-facting functionality to throw exceptions. Function does what it is asked to. Empty value = the calling code does not expect one.